Trakt/Extensions.cs

```diff
@@ -234,26 +247,166 @@ public static IList<IEnumerable<T>> ToChunks<T>(this IEnumerable<T> enumerable,
+            return results.FirstOrDefault(i => IsMatch(item, i.Show));
         }
 
         /// <summary>
-        /// Converts a progress into a splittable progress.
+        /// Gets a collected match for a series.
         /// </summary>
-        /// <param name="progress">The progress.</param>
-        /// <returns>ISplittableProgress{double}.</returns>
-        public static ISplittableProgress<double> ToSplittableProgress(this IProgress<double> progress)
+        /// <param name="item">The <see cref="Series"/>.</param>
+        /// <param name="results">IEnumerale of <see cref="TraktShowCollected"/>.</param>
+        /// <returns>TraktShowCollected.</returns>
+        public static TraktShowCollected FindMatch(Series item, IEnumerable<TraktShowCollected> results)
         {
-            var splittable = new SplittableProgress(progress.Report);
-            return splittable;
+            return results.FirstOrDefault(i => IsMatch(item, i.Show));
+        }
+
+        /// <summary>
+        /// Gets a watched match for a movie.
+        /// </summary>
+        /// <param name="item">The <see cref="BaseItem"/>.</param>
+        /// <param name="results">IEnumerale of <see cref="TraktMovieWatched"/>.</param>
+        /// <returns>TraktMovieWatched.</returns>
+        public static TraktMovieWatched FindMatch(BaseItem item, IEnumerable<TraktMovieWatched> results)
+        {
+            return results.FirstOrDefault(i => IsMatch(item, i.Movie));
+        }
+
+        /// <summary>
+        /// Gets a collected match for a movie.
+        /// </summary>
+        /// <param name="item">The <see cref="BaseItem"/>.</param>
+        /// <param name="results">IEnumerale of <see cref="TraktMovieCollected"/>.</param>
+        /// <returns>TraktMovieCollected.</returns>
+        public static IEnumerable<TraktMovieCollected> FindMatches(BaseItem item, IEnumerable<TraktMovieCollected> results)
+        {
+            return results.Where(i => IsMatch(item, i.Movie)).ToList();
+        }
+
+        /// <summary>
+        /// Checks if a <see cref="BaseItem"/> matches a <see cref="TraktMovie"/>.
+        /// </summary>
+        /// <param name="item">The <see cref="BaseItem"/>.</param>
+        /// <param name="movie">The <see cref="TraktMovie"/>.</param>
+        /// <returns><see cref="bool"/> indicating if the <see cref="BaseItem"/> matches a <see cref="TraktMovie"/>.</returns>
+        public static bool IsMatch(BaseItem item, TraktMovie movie)
+        {
+            var imdb = item.GetProviderId(MetadataProvider.Imdb);
+            if (!string.IsNullOrWhiteSpace(imdb) && string.Equals(imdb, movie.Ids.Imdb, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var tmdb = item.GetProviderId(MetadataProvider.Tmdb);
+            if (movie.Ids.Tmdb.HasValue && string.Equals(tmdb, movie.Ids.Tmdb.Value.ToString(CultureInfo.InvariantCulture), StringComparison.OrdinalIgnoreCase))
```

Comment:
I would consider changing the type of the IDs to string since we treat them as such